doc/tutorials/klondike/app/lib/step4/klondike_game.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:ui';
+
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+import 'package:flame/flame.dart';
+import 'package:flame/game.dart';
+
+import 'components/card.dart';
+import 'components/foundation_pile.dart';
+import 'components/stock_pile.dart';
+import 'components/tableau_pile.dart';
+import 'components/waste_pile.dart';
+
+class KlondikeGame extends FlameGame
+    with HasTappableComponents, HasDraggableComponents {
+  static const double cardGap = 175.0;
+  static const double cardWidth = 1000.0;
+  static const double cardHeight = 1400.0;
+  static const double cardRadius = 100.0;
+  static final Vector2 cardSize = Vector2(cardWidth, cardHeight);
+  static final cardRRect = RRect.fromRectAndRadius(
+    const Rect.fromLTWH(0, 0, cardWidth, cardHeight),
+    const Radius.circular(cardRadius),
+  );
+
+  @override
+  Future<void> onLoad() async {
+    await Flame.images.load('klondike-sprites.png');
+
+    final stock = StockPile(position: Vector2(cardGap, cardGap));
+    final waste =
+        WastePile(position: Vector2(cardWidth + 2 * cardGap, cardGap));
+    final foundations = List.generate(
+      4,
+      (i) => FoundationPile(
+        i,
+        position: Vector2((i + 3) * (cardWidth + cardGap) + cardGap, cardGap),
+      ),
+    );
+    final piles = List.generate(
+      7,
+      (i) => TableauPile(
+        position: Vector2(
+          cardGap + i * (cardWidth + cardGap),
+          cardHeight + 2 * cardGap,
+        ),
+      ),
+    );
+
+    final world = World()
+      ..add(stock)
+      ..add(waste)
+      ..addAll(foundations)
+      ..addAll(piles);
+    add(world);
+
+    final camera = CameraComponent(world: world)
+      ..viewfinder.visibleGameSize =
+          Vector2(cardWidth * 7 + cardGap * 8, 4 * cardHeight + 3 * cardGap)
+      ..viewfinder.position = Vector2(cardWidth * 3.5 + cardGap * 4, 0)
+      ..viewfinder.anchor = Anchor.topCenter;
+    add(camera);
+
+    final cards = [
+      for (var rank = 1; rank <= 13; rank++)
+        for (var suit = 0; suit < 4; suit++) Card(rank, suit)
```

Comment:
Maybe easier to read with a `...[]` surrounding Card so that it is on the next line? Or maybe harder of people don't know the spread operator...

Reply:
Even for people who do know the spread operator, what would they think when they see `...[Card(i, j)]`? I would have been wondering - why not simply write `Card(i, j)` instead? It's kinda impossible to guess that this is for the purpose of formatting...

Reply:
Yeah, it's pretty weird, but it's a quite common pattern in flutter to get the formatting to be clearer in lists, since `{}` can't be used. I'm fine with having like it is though.